services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractRegistrationService.java

```diff
@@ -192,6 +332,44 @@ public final void setTenantInformationService(final TenantInformationService ten
         return assertRegistration(tenantId, deviceId, gatewayId, NoopSpan.INSTANCE);
     }
 
+    /**
+     * Adds a new edge device in the process of auto-provisioning of not yet registered edge devices sending data
+     * via a gateway.
+     *
+     * @param tenantId The tenant that the device belongs to.
+     * @param deviceId The ID the device should be registered under.
+     * @param device Device information, must not be {@code null}.
+     * @param span The active OpenTracing span for this operation.
+     *
+     * @return A future indicating the outcome of the operation.
+     *         The <em>status code</em> is set as specified in the
+     *         <a href="https://www.eclipse.org/hono/docs/api/management/#/devices/createDeviceRegistration">
+     *         Device Registry Management API - Create Device Registration </a>
+     */
```

Comment:
As this is a abstract method and the implementations differ regarding their null checks I am not sure if it will be of any help for the reader.

Also usually I have rarely seen a NullpointerException being catched.

I think we should rather add documentation to the ``@param`` tags stating that the parameter must not be null making it part of the method's contract.

WDYT?

Reply:
This became obsolete in the refactored version.